File: social-service/src/test/java/com/eirsteir/coffeewithme/config/FriendshipControllerTestConfiguration.java
package com.eirsteir.coffeewithme.config;

import com.eirsteir.coffeewithme.social.repository.*;
import com.eirsteir.coffeewithme.social.service.coffeebreak.CoffeeBreakService;
import com.eirsteir.coffeewithme.social.service.friendship.FriendshipService;
import com.eirsteir.coffeewithme.social.service.user.UserService;
import org.mockito.Mockito;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.boot.test.mock.mockito.MockBean;
import org.springframework.context.annotation.Bean;

@TestConfiguration
public class FriendshipControllerTestConfiguration {

  @MockBean private CoffeeBreakRepository coffeeBreakRepository;

  @MockBean private UserRepository userRepository;
  @MockBean private CampusRepository campusRepository;
  @MockBean private FriendshipRepository friendshipRepository;
  @MockBean private UniversityRepository universityRepository;

  @Bean
  public FriendshipService friendshipService() {
    return Mockito.mock(FriendshipService.class);
  }

  @Bean
  public UserService userService() {
    return Mockito.mock(UserService.class);
  }

  @Bean
  public CoffeeBreakService coffeeBreakService() {
    return Mockito.mock(CoffeeBreakService.class);
  }
}

File: social-service/src/test/kotlin/com/eirsteir/coffeewithme/config/FriendshipControllerTestConfiguration.kt
package com.eirsteir.coffeewithme.config

import com.eirsteir.coffeewithme.social.service.coffeebreak.CoffeeBreakService
import org.springframework.context.annotation.Bean

@TestConfiguration
class FriendshipControllerTestConfiguration {
    @MockBean
    private val coffeeBreakRepository: CoffeeBreakRepository? = null

    @MockBean
    private val userRepository: UserRepository? = null

    @MockBean
    private val campusRepository: CampusRepository? = null

    @MockBean
    private val friendshipRepository: FriendshipRepository? = null

    @MockBean
    private val universityRepository: UniversityRepository? = null
    @Bean
    fun friendshipService(): FriendshipService? {
        return Mockito.mock(FriendshipService::class.java)
    }

    @Bean
    fun userService(): UserService? {
        return Mockito.mock(UserService::class.java)
    }

    @Bean
    fun coffeeBreakService(): CoffeeBreakService? {
        return Mockito.mock(CoffeeBreakService::class.java)
    }
}